cmd/eventlistenersink/main.go

```diff
@@ -18,200 +18,53 @@ package main
-	if sinkArgs.IsMultiNS {
-		factory = externalversions.NewSharedInformerFactory(sinkClients.TriggersClient,
-			controller.DefaultResyncPeriod)
+		log.Fatal(err.Error())
 	}
-
 	recorder, err := sink.NewRecorder()
 	if err != nil {
-		logger.Fatal(err)
-	}
-	// Create EventListener Sink
-	r := sink.Sink{
-		KubeClientSet:          kubeClient,
-		DiscoveryClient:        sinkClients.DiscoveryClient,
-		DynamicClient:          dynamicCS,
-		TriggersClient:         sinkClients.TriggersClient,
-		HTTPClient:             http.DefaultClient,
-		EventListenerName:      sinkArgs.ElName,
-		EventListenerNamespace: sinkArgs.ElNamespace,
-		PayloadValidation:      sinkArgs.PayloadValidation,
-		Logger:                 logger,
-		Recorder:               recorder,
-		Auth:                   sink.DefaultAuthOverride{},
-		WGProcessTriggers:      &sync.WaitGroup{},
-		// Register all the listers we'll need
-		EventListenerLister:         factory.Triggers().V1beta1().EventListeners().Lister(),
-		TriggerLister:               factory.Triggers().V1beta1().Triggers().Lister(),
-		TriggerBindingLister:        factory.Triggers().V1beta1().TriggerBindings().Lister(),
-		ClusterTriggerBindingLister: factory.Triggers().V1beta1().ClusterTriggerBindings().Lister(),
-		TriggerTemplateLister:       factory.Triggers().V1beta1().TriggerTemplates().Lister(),
-		ClusterInterceptorLister:    factory.Triggers().V1alpha1().ClusterInterceptors().Lister(),
-	}
-
-	startInformers()
-	// Start and sync the informers before we start taking traffic
-	withTimeout, cancel := context.WithTimeout(ctx, cacheSyncTimeout)
-	defer cancel()
-	factory.Start(ctx.Done())
-	res := factory.WaitForCacheSync(withTimeout.Done())
-	for r, hasSynced := range res {
-		if !hasSynced {
-			logger.Fatalf("failed to sync informer for: %s", r)
-		}
+		log.Fatal(err.Error())
 	}
-	logger.Infof("Synced informers. Starting EventListener")
-
-	// Listen and serve
-	logger.Infof("Listen and serve on port %s", sinkArgs.Port)
-	mux := http.NewServeMux()
-	eventHandler := http.HandlerFunc(r.HandleEvent)
-	metricsRecorder := &sink.MetricsHandler{Handler: r.IsValidPayload(eventHandler)}
-
-	mux.HandleFunc("/", http.HandlerFunc(metricsRecorder.Intercept(r.NewMetricsRecorderInterceptor())))
-
-	// For handling Liveness Probe
-	// TODO(dibyom): Livness, metrics etc. should be on a separate port
-	mux.HandleFunc("/live", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(200)
-		fmt.Fprint(w, "ok")
-	})
 
-	srv := &http.Server{
-		Addr:         fmt.Sprintf(":%s", sinkArgs.Port),
-		ReadTimeout:  sinkArgs.ELReadTimeOut * time.Second,
-		WriteTimeout: sinkArgs.ELWriteTimeOut * time.Second,
-		IdleTimeout:  sinkArgs.ELIdleTimeOut * time.Second,
-		Handler: http.TimeoutHandler(mux,
-			sinkArgs.ELTimeOutHandler*time.Second, "EventListener Timeout!\n"),
+	if sinkArgs.IsMultiNS {
+		ctx = injection.WithNamespaceScope(ctx, sinkArgs.ElNamespace)
```

Comment:
I think it should 

	if !sinkArgs.IsMultiNS {
		ctx = injection.WithNamespaceScope(ctx, sinkArgs.ElNamespace)

Because we don't want informer to be namespaced when it's multi-tenant EventListener.

Reply:
good catch!

Reply:
FWIW, this "works" since `injection.WithNamespaceScope(ctx, "")` is equivalent to not doing it, and I'm guessing there's a strong correlation between `IsMultiNS` and `ElNamespace != ""`.  I've pushed a commit to add the `!` back tho